### PR TITLE
editor use contenteditable instead of textarea

### LINKED ISSUE #97154: Use contentEditable for our hidden accessibility textArea
Investigate into using `contenteditable` instead of a plain `textArea` for our hidden accessibility field.
It migh have performance implications, but it would allow us to have richer accessibility content on each line. Including line number, breakpoints etc..

### PR DESCRIPTION
**Pull request by isidorn**
fixes #97154
Creating this PR to start discussion.

1. I have changed that the underlying element used is a `div` not a `textarea`
2. I measured performance by opening `checker.ts` and if we put 500+ line in one div the editor lags considerably. However we already have performance issues with a large text area due to chrome and thus we have a `accessibilityPageSize` setting. For the default value of this setting of 100 I could not notice any lag. 

@alexdima let me know what you think about this approach in general. If you think it makes sense I can pursue this further. I think the next steps is to allow editor decorations to be added to the hidden div element using `aria-description`. So
1) Add `ariaLabel` to the [`IModelDecorationOptions`](https://github.com/microsoft/vscode/blob/0618eaefe0979850aaa0f50b6923d80b49f3780a/src/vs/editor/common/model.ts#L75)
2) Add `textAreaHandler` api to `addAnnotation` or `addDecoration` which would add `aria-description` on the substring of the hidden div (which would have to be moved to a separate div)
3) Hoockup decorations that they call into the `textAreaHandler`
4) Line numbers could be the first client to use it, line number is just a decoration on the whole line sort of

As a consequence of this the hidden `div` can have an arbitary large dom substructure.
To make things simpler for start I suggest that we put one line in one div. And that decorations can be only put on the whole line. This will fix qiute some accessibiltiy issues.

Once we have that dom substructure per line we can look into measuring performance.

### CODE REVIEW

**Comment by isidorn** [comment]
The `textAreaState` has no notion of `lineNumbers`, so it might be easier to not even introduce the notion of line numbers there. But just to allow decorations to be put on a substring based on offset.
The ugly part there is that since we only put a subset of the whole content (due to performance reasons) as the user scroll we will have to re-set the decorations.

**Comment by isidorn** [comment]
Feedback from Alex

 I did a quick look last week and it makes total sense what you are doing.
The very difficult thing is that in order to actually merge this in we need a lot of testing. The input of the editor depends on browser, OS, and keyboard layout / IME. So there is a huge test matrix and IMHO we would need to do a whole lot of testing to have this be the default, because maybe on Safari/iOS the contenteditable behaves slightly different than a textarea, etc. Some of the input ways we know about are documented here -- https://github.com/microsoft/vscode/wiki/IME-Test -- but there might be many more ways which work today and are just not documented there...

An approach would be to have a setting and have both code paths (with contenteditable or with the textarea based on the setting). I am a bit defensive to merge this unless it is behind a setting because we might break some huge population doing some kind of user input which we don't know about. There are other factors, where I heard in the past that in the content editable it is possible to drag&drop images and I see you now use `.textContent` to replace the `textarea.value`, but I don't know if that always works OK.

TL;DR: Looks good for exploring, but maybe it needs to be implemented (at least initially) behind a setting since I'm so worried with wide user input breakage which cannot be covered by unit tests... This is a surprisingly deep topic -- look at the amount of bugs we get in this area -- https://github.com/microsoft/vscode/issues?q=is%3Aopen+is%3Aissue+label%3Aeditor-input-IME
